js/main.js

```diff
@@ -42,26 +45,22 @@ $form.addEventListener('submit', function saveEntry(event) {
     for (let i = 0; i < data.entries.length; i++) {
       if (data.entries[i].entryId === editInput.entryId) {
         data.entries[i] = editInput;
-      }
-    }
-    // add the edited form to the DOM tree using renderEntry function and find the list element (the entry form) that needs to be replaced with the updated version and replace it
-    const $updatedForm = renderEntry(editInput);
-    const $liElements = document.querySelectorAll('li');
-    for (let i = 0; i < $liElements.length; i++) {
-      if (editInput.entryId === Number($liElements[i].getAttribute('data-entry-id'))) {
+        const $updatedForm = renderEntry(editInput);
         $liElements[i].replaceWith($updatedForm);
       }
     }
   }
   // reset the form and switch to entries view to show the updated forms after the edits with the rest of the entries
   $form.reset();
+  data.editing = null;
+  toggleNoEntries();
+  data.view = 'entries';
```

Comment:
You shouldn't be updating `data.view` directly here, this will get updated inside of your `viewSwap` function